### PR TITLE
Add payload size to messaging semantic conventions

### PR DESCRIPTION
**Pull request by arminru**
I forgot about that one on my previous PR, could be useful as well.

### CODE REVIEW

File: specification/trace/semantic_conventions/messaging.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ The processor of the message should set the kind to `CONSUMER`, unless it always
 | `messaging.url` | Connection string such as `tibjmsnaming://localhost:7222` or `https://queue.amazonaws.com/80398EXAMPLE/MyQueue`. | No |
 | `messaging.message_id` | A value used by the messaging system as an identifier for the message, represented as a string. | No |
 | `messaging.conversation_id` | A value identifying the conversation to which the message belongs, represented as a string. Sometimes called "Correlation ID". | No |
+| `messaging.message_payload_size_bytes` | The size of the message's payload in bytes. | No |
```
Comment (bogdandrutu) [inline specification/trace/semantic_conventions/messaging.md:65]: Is this size compressed or uncompressed?

Comment (arminru) [reply specification/trace/semantic_conventions/messaging.md:65]: Compression doesn't apply to all systems or client libraries and some won't even specify which size they return (like [IBM MQ](https://www.ibm.com/support/knowledgecenter/en/SSFKSJ_9.0.0/com.ibm.mq.javadoc.doc/WMQJavaClasses/com/ibm/mq/MQMessage.html?view=embed#getMessageLength--), for example). I'd either keep that open as it is or introduce two separate attributes (`_compressed` and `_uncompressed`) but that would leave users who don't know which size is returned by the library to guess which attribute they should use. What would you prefer?

Comment (bogdandrutu) [reply specification/trace/semantic_conventions/messaging.md:65]: We decided in OC to go with to, and when in doubt we suggested users to use "uncompressed" :)

Comment (arminru) [reply specification/trace/semantic_conventions/messaging.md]: Sound reasonable, I've split it up.

File: specification/trace/semantic_conventions/messaging.md [hunk 2]
```diff
@@ -62,11 +62,15 @@ The processor of the message should set the kind to `CONSUMER`, unless it always
 | `messaging.url` | Connection string such as `tibjmsnaming://localhost:7222` or `https://queue.amazonaws.com/80398EXAMPLE/MyQueue`. | No |
 | `messaging.message_id` | A value used by the messaging system as an identifier for the message, represented as a string. | No |
 | `messaging.conversation_id` | A value identifying the conversation to which the message belongs, represented as a string. Sometimes called "Correlation ID". | No |
+| `messaging.message_payload_size_uncompressed_bytes` | The uncompressed size of the message's payload in bytes. | No |
```
Comment (Oberon00) [inline specification/trace/semantic_conventions/messaging.md:65]:
```suggestion
| `messaging.message_payload_size_uncompressed_bytes` | The uncompressed size of the message payload in bytes. | No |
```

To be consistent with "message destination", etc.

**Comment by Oberon00** [comment]
As @bogdandrutu [suggested](https://github.com/open-telemetry/opentelemetry-specification/pull/550#discussion_r405836035):

> when in doubt we suggested users to use "uncompressed"

Should we add a note about that here?

**Comment by arminru** [comment]
@Oberon00
That note was already there but since it's apparently too easy to overlook below I placed it more prominently in 3cda5cb 🙂

File: specification/trace/semantic_conventions/messaging.md [hunk 3]
```diff
@@ -62,6 +62,8 @@ The processor of the message should set the kind to `CONSUMER`, unless it always
 | `messaging.url` | Connection string such as `tibjmsnaming://localhost:7222` or `https://queue.amazonaws.com/80398EXAMPLE/MyQueue`. | No |
 | `messaging.message_id` | A value used by the messaging system as an identifier for the message, represented as a string. | No |
 | `messaging.conversation_id` | A value identifying the conversation to which the message belongs, represented as a string. Sometimes called "Correlation ID". | No |
+| `messaging.message_payload_size_uncompressed_bytes` | The uncompressed size of the message payload in bytes. Also use this attribute if it is unknown whether the compressed or uncompressed payload size is reported. | No |
```
Comment (bogdandrutu) [inline specification/trace/semantic_conventions/messaging.md:65]: Maybe just drop the `uncompressed` to support more that this is default when the information is not known?

Comment (arminru) [reply specification/trace/semantic_conventions/messaging.md]: Sounds good, done!

**Pull request merged**